main.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    item_select = []
+    my_each { |i| item_select << i if yield(i) }
+    item_select
+  end
+
+  def my_all?
+    return my_each { |i| return false if yield(i) == false } unless block_given?
+
+    true
+  end
+
+  def my_any?
+    return my_each { |i| return true if yield(i) } unless block_given?
+
+    false
+  end
+
+  def my_none?
+    return my_each { |i| return false if yield(i) } unless block_given?
+
+    true
+  end
+
+  def my_count(items = nil)
+    count = 0
+    if block_given?
+      my_each { |i| count += 1 if yield(i) == true }
+    elsif items.nil?
+      my_each { count += 1 }
+    else
+      my_each { |i| count += 1 if i == items }
+    end
+    count
+  end
+
+  def my_map(arg = nil)
+    return to_enum unless block_given?
+
+    arr = []
+    my_each do |i|
+      arr << if !arg.nil?
+               arg.call(i)
+             else
+               yield(i)
+             end
+    end
+    arr
+  end
+
+  def my_inject(value = self[0])
+    value ||= []
+    my_each do |element|
+      value = yield(value, element)
+    end
+    value
+  end
+end
```

Comment:
Your my_inject method needs some fixing :bug: 
enumerables my_inject combines all elements of enum by applying a binary operation, specified by a block
      Failure/Error: yield(self[i])
      
      NoMethodError:
        undefined method `[]' for 5..50:Range

Reply:
Hello, @Ispirett I have corrected the errors you highlighted and submitted for another review.